sentry_streams/tests/adapters/arroyo/test_consumer.py

```diff
@@ -94,3 +101,71 @@ def test_single_route(broker: LocalBroker[KafkaPayload], pipeline: Pipeline) ->
             call({Partition(Topic("logical-events"), 0): 4}),
         ]
     )
+
+
+def test_reduce(broker: LocalBroker[KafkaPayload], reduce_pipeline: Pipeline) -> None:
+    """
+    Test the creation of an Arroyo Consumer from a number of
+    pipeline steps.
+    """
+
+    consumer = ArroyoConsumer(source="source1")
+    consumer.add_step(
+        MapStep(
+            route=Route(source="source1", waypoints=[]),
+            pipeline_step=cast(Map, reduce_pipeline.steps["decoder"]),
+        )
+    )
+    consumer.add_step(
+        MapStep(
+            route=Route(source="source1", waypoints=[]),
+            pipeline_step=cast(Map, reduce_pipeline.steps["mymap"]),
+        )
+    )
+    consumer.add_step(
+        ReduceStep(
+            route=Route(source="source1", waypoints=[]),
+            pipeline_step=cast(Reduce[timedelta, Any, Any], reduce_pipeline.steps["myreduce"]),
+        )
+    )
+    consumer.add_step(
+        StreamSinkStep(
+            route=Route(source="source1", waypoints=[]),
+            producer=broker.get_producer(),
+            topic_name="transformed-events",
+        )
+    )
+
+    factory = ArroyoStreamingFactory(consumer)
+    commit = mock.Mock(spec=Commit)
+    strategy = factory.create_with_partitions(commit, {Partition(Topic("logical-events"), 0): 0})
+
+    strategy.submit(make_msg("msg1", "logical-events", 0))  # first message
+    time.sleep(2)
+    strategy.submit(make_msg("msg2", "logical-events", 1))  # second message in the first batch
+    time.sleep(2)
+    strategy.submit(make_msg("msg3", "logical-events", 2))  # second message in the first batc
+    time.sleep(3)
+    strategy.poll()
+
+    topic = Topic("transformed-events")
+    msg1 = broker.consume(Partition(topic, 0), 0)
+    assert msg1 is not None and msg1.payload.value == "msg1_mappedmsg2_mappedmsg3_mapped".encode(
+        "utf-8"
+    )
+
+    time.sleep(2)
```

Comment:
Can't you mock the timestamp passed to the submit function to simulate windows ?